hudi-spark2/src/main/java/org/apache/hudi/DataSourceUtilsForSpark2.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi;
+
+import org.apache.hudi.common.model.HoodieTableType;
+import org.apache.hudi.common.model.WriteOperationType;
+import org.apache.hudi.common.table.timeline.HoodieTimeline;
+import org.apache.hudi.common.util.CommitUtils;
+import org.apache.hudi.config.HoodieCompactionConfig;
+import org.apache.hudi.config.HoodieIndexConfig;
+import org.apache.hudi.config.HoodieWriteConfig;
+import org.apache.hudi.index.HoodieIndex;
+
+import java.util.Map;
+
+/**
+ * Utilities used throughout the data source.
+ * TODO: This file is partially copied from org.apache.hudi.DataSourceUtils.
+ * Should be removed if Spark 2.x support is dropped.
```

Comment:
would be good to have a JIRA with all these follow ups when Spark 2.x support is dropped

Reply:
I created a new module ```hudi-spark-common``` to avoid creating xxxForSpark2 classes. 
So the current structure of hudi-spark would be:

hudi-spark-datasource
                 |
                 -----------------hudi-spark
                 |
                -----------------hudi-spark-common
                 |
                -----------------hudi-spark2
                 |
                -----------------hudi-spark3